### PR TITLE
Feature/add release event publishing

### PR DESCRIPTION
**Pull request by kjuulh**
This is the first step of adding events to be processed both internally and externally.

We'll start with release event

@MariaCFFrandsen maybe you would like to comment on the model, some of the fields you mentioned, sadly isn't possible. But I've taken the liberty to add a few more.

### CODE REVIEW

File: cmd/server/command/start.go [hunk 1]
```diff
@@ -249,37 +280,65 @@ func NewStart(startOptions *startOptions) *cobra.Command {
...
+					metricsObserver.ObserveRelease(
+						metrics.Release{
+							Environment: opts.Environment,
+							Service:     opts.Service,
+							Releaser:    opts.Releaser,
+							Intent:      opts.Intent.Type,
+						},
+					)
+				},
+				"events": func(ctx context.Context, opts flow.NotifyReleaseOptions) {
```
Comment (MariaCFFrandsen) [inline cmd/server/command/start.go:304]: @kjuulh Looks good to me - Thanks for spending time on it!

I have some questions:)
It means that the queue defaults to "release-manager" - do you know if this is true? And do you know where the value is retrieved from on start https://github.com/lunarway/release-manager/blob/master/cmd/server/command/root.go#L129?

release-manager uses the github.com/rabbitmq/amqp091-go directly. 
When we need to consume, would be possible to go through go.lunarway.com/amqp and what approach would you recommend?

Comment (kjuulh) [reply cmd/server/command/start.go:304]: We cannot go through go.lunarway.com/amqp as that is a private library. You would need to listen on the exchange in the same manner as we do in release manager it self

Comment (MariaCFFrandsen) [reply cmd/server/command/start.go:304]: No, but we can in development-metrics :) Alright.

File: cmd/server/command/start.go [hunk 2]
```diff
@@ -249,37 +280,65 @@ func NewStart(startOptions *startOptions) *cobra.Command {
...
 						Service:     opts.Service,
-						Releaser:    opts.Releaser,
-						Intent:      opts.Intent.Type,
+						Namespace:   opts.Namespace,
+						ArtifactID:  opts.Spec.ID,
+						AuthorEmail: opts.Spec.Application.AuthorEmail,
+						AuthorName:  opts.Spec.Application.AuthorName,
+						Environment: opts.Environment,
+						Squad:       opts.Spec.Squad,
+						IntentType:  opts.Intent.Type,
```
Comment (Crevil) [inline cmd/server/command/start.go:314]: I think some of the intent details are valuable to provide as well, eg. if intent is `Rollback` the `PreviousArtifactID` is nice to know "from where" the rollback happened.

@MariaCFFrandsen has some ideas here that we will flesh out tomorrow in a DDD session.

File: cmd/server/command/start.go [hunk 3]
```diff
@@ -249,37 +280,65 @@ func NewStart(startOptions *startOptions) *cobra.Command {
...
 					})
+
+					if err != nil {
+						logger.Errorf("could not publish released_event. Error: %w", err)
+					} else {
+						logger.Debugf("published released_event for artifactId: %s", opts.Spec.ID)
+					}
```
Comment (Crevil) [inline cmd/server/command/start.go:321]: We could drop the `else` here as we tend to in Go 🙈 

```suggestion
						return
					}
					logger.Debugf("published released_event for artifactId: %s", opts.Spec.ID)
```

File: cmd/server/command/start.go [hunk 4]
```diff
@@ -309,26 +367,40 @@ func NewStart(startOptions *startOptions) *cobra.Command {
 					}
 					return flowSvc.ExecNewArtifact(context.Background(), event)
 				},
+				events.ReleasedEvent{}.Type(): func(d []byte) error {
+					var event events.ReleasedEvent
```
Comment (Crevil) [inline cmd/server/command/start.go:371]: Do we need to add a handler for it?

**Review by MariaCFFrandsen — APPROVED** [review]

File: cmd/server/command/start.go (hunk 1 above)
Comment (kjuulh) [reply cmd/server/command/start.go:304]: I will add poderror event when we've gotten your subscription stuff to work

**Review by Crevil — APPROVED** [review]
Overall I think this is sound. There are _a lot_ of unrelated changes in the PR which makes it hard to read. I guess the functional change is really just half the changed lines if not less 🙈

**Comment by kjuulh** [comment]
Yeah much less we probably need a full gofmt pr just so that we don't have to bother each following pr

**Pull request merged**